src/Pipeline.cpp

```diff
@@ -565,25 +591,54 @@ void Pipeline::compile_jit(const Target &target_arg) {
                  << target << "\n";
         return;
     }
-
     // Clear all cached info in case there is an error.
     contents->invalidate_cache();
 
     // Infer an arguments vector
-    infer_arguments();
-
+    if (!debug_serialization.empty()) {
+        deserialized_pipe.infer_arguments();
+        // infer_arguments();
```

Comment:
As always, commented out code should be removed or have a comment, even for a draft PR :-)